x-pack/plugins/cloud_security_posture/server/routes/compliance_dashboard/get_clusters.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from 'kibana/server';
+import {
+  AggregationsMultiBucketAggregateBase as Aggregation,
+  SearchRequest,
+} from '@elastic/elasticsearch/lib/api/types';
+import { CloudPostureStats } from '../../../common/types';
+import {
+  getResourceTypeFromAggs,
+  resourceTypeAggQuery,
+  type ResourceTypeQueryResult,
+} from './get_resources_types';
+import { CSP_KUBEBEAT_INDEX_PATTERN } from '../../../common/constants';
+import { findingsEvaluationAggsQuery, getStatsFromFindingsEvaluationsAggs } from './get_stats';
+import { KeyDocCount } from './compliance_dashboard';
+
+export interface ClusterBucket extends ResourceTypeQueryResult, KeyDocCount {
+  failed_findings: {
+    doc_count: number;
+  };
+  passed_findings: {
+    doc_count: number;
+  };
+  benchmarks: Aggregation<KeyDocCount>;
+}
+
+interface ClustersQueryResult {
+  aggs_by_cluster_id: Aggregation<ClusterBucket>;
+}
+
+export const getClustersQuery = (cycleId: string): SearchRequest => ({
+  index: CSP_KUBEBEAT_INDEX_PATTERN,
+  size: 0,
+  query: {
+    bool: {
+      filter: [{ term: { 'cycle_id.keyword': cycleId } }],
+    },
+  },
+  aggs: {
+    aggs_by_cluster_id: {
+      terms: {
+        field: 'cluster_id.keyword',
+      },
+      aggs: {
+        benchmarks: {
+          terms: {
+            field: 'rule.benchmark.keyword',
+          },
+        },
+        ...resourceTypeAggQuery,
+        ...findingsEvaluationAggsQuery,
+      },
+    },
+  },
+});
+
+export const getClustersFromAggs = (clusters: ClusterBucket[]): CloudPostureStats['clusters'] =>
+  clusters.map((cluster) => {
+    // get cluster's meta data
+    const benchmarks = cluster.benchmarks.buckets;
+    if (!Array.isArray(benchmarks)) throw new Error('missing aggs by benchmarks per cluster');
+
+    const meta = {
+      clusterId: cluster.key,
+      benchmarkName: benchmarks[0].key,
+    };
+
+    // get cluster's stats
+    if (!cluster.failed_findings || !cluster.passed_findings)
+      throw new Error('missing findings evaluations per cluster');
```

Comment:
`import {number} from 'io-ts'` 

```suggestion
    if (!number.is(cluster.failed_findings) || !number.is(cluster.passed_findings))
      throw new Error('missing findings evaluations per cluster');
```

in its current form, if we have `0` failed or passed findings (which is a valid state - all failed / all passed) - this will throw

Reply:
i totally get why you would think that but in this case those are just the field names, which are typed. `cluster` looks like this:
```
cluster: {
failed_findings: {
doc_count: number
},
passed_findings: {
doc_count: number
}
// more fields
}
```